framework/app.mk

```diff
@@ -159,13 +159,14 @@ endif
 
 BUILD_EXEC :=
 
-$(app_objects): $(app_HEADER)
-
+app_GIT_DIR := $(shell cd "$(APPLICATION_DIR)" && git rev-parse --show-toplevel)
```

Comment:
If the git command fails then the error will show up while doing a make.
Since we do a `wildcard` for the actual deps, the files won't be found and then `app_HEADER_deps` will be empty.
The revision file should still be generated but obviously won't ever get updated.